packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -128,6 +135,48 @@ class UnpackMacOS extends Target {
   }
 }
 
+/// Unpack the release prebuilt engine framework.
+class ReleaseUnpackMacOS extends UnpackMacOS {
+  const ReleaseUnpackMacOS();
+
+  @override
+  String get name => 'release_unpack_macos';
+
+  @override
+  List<Source> get inputs => const <Source>[
+    Source.pattern('{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/macos.dart'),
```

Comment:
Could this source be a getter in the base class rather than hard-coded in all the subclasses? Alternately the parent class could implement inputs by adding this input to the result of a new abstract method (e.g. `modeSpecificInputs`) that the subclasses implement. (Or even just have the build mode as a getter in the subclasses since the lookup logic is otherwise the same.)

Reply:
I've deduped the inputs/outputs into the base class.

Reply:
Is there an unpushed change? It looks the same here.

Reply:
Lost somewhere, re-fixed